src/lib.rs

```diff
@@ -851,103 +854,272 @@ pub struct RegisteredClaims {
+///
+/// Should any temporal claims be needed, set the appropriate fields.
+///
+/// To deal with clock drifts, you might want to provide an `epsilon` error margin in the form of a
+/// `chrono::Duration` to allow time comparisons to fall within the margin.
+pub struct TemporalOptions {
+    /// Allow for some leeway for clock drifts, limited to this duration during temporal validation
+    pub epsilon: Duration,
+
+    /// Specify a time to use in temporal validation instead of `Now`
+    pub now: Option<DateTime<Utc>>,
+}
+
+impl Default for TemporalOptions {
+    fn default() -> Self {
+        TemporalOptions {
+            epsilon: Duration::seconds(0),
+            now: None,
+        }
+    }
+}
+
+#[derive(Eq, PartialEq, Clone)]
+/// Options for claims validation
+///
+/// If a claim is missing, it passes validation unless the claim is marked as required within the
+/// `claim_presence_options`.
+///
+/// By default only expiry-related claims are validated (namely `exp`, `nbf`, `iat`).
+///
+///
+/// are required, and they pass validation if they are missing.
 ///
 /// Should any temporal claims be needed, set the appropriate fields.
 ///
 /// To deal with clock drifts, you might want to provide an `epsilon` error margin in the form of a
 /// `std::time::Duration` to allow time comparisons to fall within the margin.
-pub struct TemporalValidationOptions {
+pub struct ValidationOptions {
+    ///Claims marked as required will trigger a validation failure if they are missing
+    pub claim_presence_options: ClaimPresenceOptions,
+
+    ///Define how to validate temporal claims
+    pub temporal_options: TemporalOptions,
+
     /// Whether the `iat` or `Issued At` field is required
-    pub issued_at_required: bool,
+    /// Parameter shows the maximum age of a token to be accepted.
+    pub issued_at: Validation<Duration>,
     /// Whether the `nbf` or `Not Before` field is required
-    pub not_before_required: bool,
+    pub not_before: Validation<()>,
     /// Whether the `exp` or `Expiry` field is required
-    pub expiry_required: bool,
-    /// Allow for some clock drifts, limited to this duration during temporal validation
-    pub epsilon: Option<std::time::Duration>,
-    /// Specify a time to use in temporal validation instead of `Now`.
-    pub now: Option<DateTime<Utc>>,
+    pub expiry: Validation<()>,
+
+    /// Whether the `iss` or `Issuer` field is required
+    /// Parameter must match the issuer in the token exactly.
+    pub issuer: Validation<StringOrUri>,
+
+    /// Whether the `aud` or `Audience` field is required
+    /// Token must include an audience with the value of the parameter
+    pub audience: Validation<StringOrUri>,
+    //    pub subject_validated: Validation<Box<FnOnce(String) -> bool>>,
+    //    pub custom_validation: Validation<Box<FnOnce(&RegisteredClaims) -> Result<(), ValidationError>>>
+}
+
+impl Default for ValidationOptions {
+    fn default() -> Self {
+        ValidationOptions {
```

Comment:
This whole block can be simplified:

```rust
ValidationOptions {
  issued_at: Validation::Validate((Default::default(), Duration::max_value())),
  ..Default::default()
}
```

Reply:
Well `expiry` and `not_before` also have non-default values, and I wanted the full list here so that anyone browsing through the code will see instantly what the default values are without having to go deeper